server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraExtension.java

```diff
@@ -56,4 +59,14 @@ public void pause() {
     public void unpause() {
         cassandra.unpause();
     }
+
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return parameterContext.getParameter().getType() == Host.class;
```

Comment:
I'm suspicious here...

What if we add the `Host` resolution capability to other backends (akka ElasticSearch?)

Let me suggest to add  a `@Cassandra` annotation that we can access via `ParameterContext` to disambiguate that!

Reply:
I made it to `CassandraHost` to be more clear